### PR TITLE
binance: add futures stream

### PR DESCRIPTION
**Pull request by austin362667**
not implement yet:

1. OrderTradeUpdateEvent `case "TRADE"`
2. OrderTradeUpdateEvent `case "CALCULATED - Liquidation Execution"`

### CODE REVIEW

File: pkg/exchange/binance/stream.go [hunk 1]
```diff
@@ -211,6 +213,24 @@ func NewStream(ex *Exchange, client *binance.Client, futuresClient *futures.Clie
...
+		positionSnapshot := types.PositionMap{}
+		positionSnapshot = toGlobalFuturesPositions(e.AccountUpdate.Positions)
+		log.Infof("positionMap: %+v",
+			positionSnapshot)
+		stream.EmitPositionSnapshot(positionSnapshot)
+
+		balanceSnapshot := types.BalanceMap{}
+		balanceSnapshot = toGlobalFuturesBalance(e.AccountUpdate.Balances)
+		log.Infof("balanceMap: %+v",
+			balanceSnapshot)
```
Comment (c9s) [inline pkg/exchange/binance/stream.go:227]: remove debug message?

Comment (austin362667) [reply pkg/exchange/binance/stream.go]: sorry.. removed

File: pkg/exchange/binance/stream.go [hunk 2]
```diff
@@ -158,6 +160,20 @@ func NewStream(ex *Exchange, client *binance.Client, futuresClient *futures.Clie
 	stream.OnExecutionReportEvent(stream.handleExecutionReportEvent)
 	stream.OnContinuousKLineEvent(stream.handleContinuousKLineEvent)
 
+	// Event type ACCOUNT_UPDATE from user data stream updates Balance and FuturesPosition.
+	stream.OnAccountUpdateEvent(func(e *AccountUpdateEvent) {
+		futuresPositionSnapshot := types.FuturesPositionMap{}
+		futuresPositionSnapshot = toGlobalFuturesPositions(e.AccountUpdate.Positions)
```
Comment (c9s) [inline pkg/exchange/binance/stream.go:166]: you can just use `:=` here?

File: pkg/exchange/binance/stream.go [hunk 3]
```diff
@@ -158,6 +160,20 @@ func NewStream(ex *Exchange, client *binance.Client, futuresClient *futures.Clie
...
+		stream.EmitFuturesPositionSnapshot(futuresPositionSnapshot)
+
+		balanceSnapshot := types.BalanceMap{}
+		balanceSnapshot = toGlobalFuturesBalance(e.AccountUpdate.Balances)
```
Comment (c9s) [inline pkg/exchange/binance/stream.go:170]: also here~

File: pkg/exchange/binance/stream.go [hunk 4]
```diff
@@ -158,6 +160,20 @@ func NewStream(ex *Exchange, client *binance.Client, futuresClient *futures.Clie
...
+		stream.EmitBalanceSnapshot(balanceSnapshot)
+	})
+
+	// TODO: emit account config leverage updates
+	stream.OnAccountConfigUpdateEvent(func(e *AccountConfigUpdateEvent) {})
```
Comment (c9s) [inline pkg/exchange/binance/stream.go:175]: add a handleAccountConfigUpdateEvent method for registering callback

File: pkg/exchange/binance/stream.go [hunk 5]
```diff
@@ -158,6 +160,20 @@ func NewStream(ex *Exchange, client *binance.Client, futuresClient *futures.Clie
 	stream.OnExecutionReportEvent(stream.handleExecutionReportEvent)
 	stream.OnContinuousKLineEvent(stream.handleContinuousKLineEvent)
 
+	// Event type ACCOUNT_UPDATE from user data stream updates Balance and FuturesPosition.
+	stream.OnAccountUpdateEvent(func(e *AccountUpdateEvent) {
```
Comment (c9s) [inline pkg/exchange/binance/stream.go:164]: move this function to `handleOnAccountUpdateEvent(e ...)` ?

**Review by c9s — APPROVED** [review]

**Pull request merged**